client/src/components/HomePage/modal/JoinModal/index.jsx

```diff
@@ -33,17 +38,21 @@ class Join extends Component {
             context,
           );
         } else if (!validator.equals(password, confirmPassword)) {
-          this.setPopUp(
-            'Passwords do not match',
-            'Error !',
-            context,
-          );
+          this.setPopUp('Passwords do not match', 'Error!', context);
         } else {
-          this.setState({ signingUp: false });
-          context.updateState({ completeJoin: true });
+          axios
+            .get('/api/v1/signup', { headers: { email } })
+            .then(() => {
+              this.setState({ signingUp: false });
+              context.updateState({ completeJoin: true });
```

Comment:
@HemaSAli for the user, what happens next? Should this redirect them to another page after a successful signup?

Reply:
It will redirect him to first Stage of details, btw. we have two sages about details, First stage is about his language and the second is about his job
so ```context.updateState({ completeJoin: true });``` means than he complete the main stage which contain his email, first, last name and his password